internal/controllers/machinehealthcheck/machinehealthcheck_controller.go

```diff
@@ -539,6 +539,14 @@ func (r *Reconciler) patchUnhealthyTargets(ctx context.Context, logger logr.Logg
 			klog.KObj(t.Machine),
 			klog.KObj(t.MHC),
 		)
+		// Record event on MachineHealthCheck to track remediation history
+		r.recorder.Eventf(
+			m,
+			corev1.EventTypeNormal,
+			EventRemediationTriggered,
+			"Triggered remediation for unhealthy Machine %s",
```

Comment:
Remediation is not triggered by MHC, MHC only marks the machine as unhealthy (as well as it unmarks it in case it goes back to healthy).

Also, I would not add the name of the machine to an event added to the machine itself, it is redundant. May be, let's consider instead to track which MHC instance applied the event